File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int n=prices.size();
       int mini=prices[0],maxProfit=0;
       int currProfit;
       for(int i=1;i<n;i++) {
           mini=min(mini,prices[i]);
            currProfit=prices[i]-mini;
            maxProfit=max(maxProfit,currProfit);
       }
       return maxProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
                
        if(prices.length == 0){
            return 0;
        }
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for(int price:prices){
            if(price - minPrice > maxProfit){
                maxProfit = price-minPrice;
            } else if(price < minPrice) {
                minPrice = price;
            }
        }

        return maxProfit;
    }
}